src/braket/devices/local_simulator.py

```diff
@@ -98,8 +88,12 @@ def run(
             >>> device = LocalSimulator("default")
             >>> device.run(circuit, shots=1000)
         """
-        result = self._run_internal(task_specification, shots, inputs=inputs, *args, **kwargs)
-        return LocalQuantumTask(result)
+        args = _convert_to_sim_format(
+            task_specification, self._delegate, shots, inputs=inputs
+        ) + list(args)
+        execute_manager = self._delegate.execute_manager(*args, **kwargs)
+
+        return LocalQuantumTask.create(execute_manager, *args, **kwargs)
```

Comment:
What if a device only allows running programs synchronously (i.e `.execute_manger` isn't implemented)?

Reply:
Follow up comment from `hasattr`. User has to define execute manager. Even if the simulator only supports synchronously we run that on a different thread. Once we call `result` on `LocalQuantumTask`. It will wait on the thread and it will block the execution.

please correct me, if this isn't expected